datalake/src/main/java/com/sequenceiq/datalake/service/sdx/StackRequestManifester.java

```diff
@@ -269,6 +270,27 @@ void setupCloudStorageAccountMapping(StackV4Request stackRequest, String environ
         }
     }
 
+    void validateMappingsConfig(MappingsConfig mappingsConfig, StackV4Request stackRequest) {
+        String errorMessage = null;
+
+        // Validate RAZ if enabled, making sure that the RAZ mapping exists only when it is required.
+        if (stackRequest.getCluster().isRangerRazEnabled()) {
+            if (!mappingsConfig.getActorMappings().containsKey("rangerraz")) {
+                errorMessage = "IDBMMS mappings must contain the RAZ role if RAZ is to be created!";
+            }
+        } else {
+            if (mappingsConfig.getActorMappings().containsKey("rangerraz")) {
```

Comment:
I see the general intent of this check, but IDBMMS always includes mappings for other service principals even if they will never be used by any clusters. So I see no real benefit of prohibiting this mapping rule if no RAZ is to be installed. (No other users are supposed to be able to ever impersonate themselves as `rangerraz`.)

Reply:
+1 to @lajosrodek's comment. I was the one who initially suggested that we might want to disallow setting the RAZ role when creating non-RAZ datalakes, but Lajos makes a good point here, which hadn't occurred to me.

The rangerraz role is a more powerful role than other roles, but, as Lajos points out, our security model assumes that users are not able to impersonate other users.